shardy/dialect/sdy/ir/attrs.td

```diff
@@ -746,6 +793,20 @@ def Sdy_OpShardingRule : AttrDef<Sdy_Dialect, "OpShardingRule"> {
     these ops, so a user must tell it how. When it is a custom rule, then the
     rule is always preserved/never removed. `is_custom_rule` can only be true
     for `stablehlo.custom_call` ops.
+
+    **Constraints:**
+    - Number of operand/result mappings must match the number of
+      operands/results of the op.
+    - There is at least one mapping (can't have a rule for an op with no
+      operands/results).
+    - Rank of each `TensorMappingAttr` matches the rank of the corresponding
```

Comment:
why don't we have constraints for TensorMappingAttr? Also we don't refer to constraints in DimMappingAttr anywhere, we should for every element in `dim_mapping` in `TensorMappingAttr`

wdyt?

Reply:
Probably because I misunderstood the code. I only see verifiers for ShardingRuleMapping and not for TensorMapping - I will add something.